### PR TITLE
CLOUDSTACK-10365: Change the "getXXX" boolean-related method names to…

### PR DESCRIPTION
**Pull request by BruceKuiLiu**
… "isXXX".

These boolean-return methods are named as "getXXX".
Other boolean-return methods are named as "isXXX".
Considering there methods will return boolean values, it should be more clear and consistent to rename them as "isXXX".

### CODE REVIEW

File: api/src/main/java/com/cloud/agent/api/to/S3TO.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ public DataStoreRole getRole() {
         return DataStoreRole.Image;
     }
 
-    public boolean getEnableRRS() {
+    public boolean isEnableRRS() {
```
Comment (DaanHoogland) [inline api/src/main/java/com/cloud/agent/api/to/S3TO.java]: what is the is prefix for in this case? is the question enableRRS() ?

File: api/src/main/java/com/cloud/offering/ServiceOffering.java [hunk 2]
```diff
@@ -76,7 +76,7 @@
     /**
      * @return Does this service plan offer HA?
      */
-    boolean getOfferHA();
+    boolean isOfferHA();
```
Comment (DaanHoogland) [inline api/src/main/java/com/cloud/offering/ServiceOffering.java]: the java doc implies it should be called doesOfferHA()

File: api/src/main/java/com/cloud/offering/ServiceOffering.java [hunk 3]
```diff
@@ -86,7 +86,7 @@
     /**
      * @return Does this service plan support Volatile VM that is, discard VM's root disk and create a new one on reboot?
      */
-    boolean getVolatileVm();
+    boolean isVolatileVm();
```
Comment (DaanHoogland) [inline api/src/main/java/com/cloud/offering/ServiceOffering.java]: supportVolitileVMs() would be more pleasing on the human eye. Not that that should hamper this change but what is the objective? to adhere to another naming convention or to make better code for reading? and seriously both are valid motives under circumstances, but I am really wondering.

Comment (BruceKuiLiu) [reply api/src/main/java/com/cloud/offering/ServiceOffering.java]: The intention of my pull requests is to make code more readable and understandable, thus further making developers to understand the code more easily with method names without more context information. 
But about method naming things, developers have two different opinions as you mentioned:
(1) keep the same naming convention as similar methods.
(2) make code more readable.

Comment (DaanHoogland) [reply api/src/main/java/com/cloud/offering/ServiceOffering.java]: @BruceKuiLiu I am with 2 all the way, with the exception of the use of these methods in code generation tools. In this last method I would expect a name like isForVolatileVms() or the ealier sugested name. Of cause the accessor is named after the field implying that the field is not properly named either.
I am +0 on this one. Try to find more reviews.

File: api/src/main/java/com/cloud/offering/ServiceOffering.java (hunk 2 above)
Comment (fmaximus) [reply api/src/main/java/com/cloud/offering/ServiceOffering.java]: @DaanHoogland I would prefer to only have methods starting with either get or is.
Any other method name in a VO would not be usable for filtering,
according to the implementation of the VO MethodInterceptor

Comment (DaanHoogland) [reply api/src/main/java/com/cloud/offering/ServiceOffering.java]: good point @fmaximus so the generic dao breaks if we go any other way is your message, is it?

Comment (rafaelweingartner) [reply api/src/main/java/com/cloud/offering/ServiceOffering.java]: Do you know the name of the class that does this interception, so we can inspect it?
Is it using method names or POJO's properties directly via their annotation?

File: api/src/main/java/com/cloud/agent/api/to/S3TO.java (hunk 1 above)
Comment (rafaelweingartner) [reply api/src/main/java/com/cloud/agent/api/to/S3TO.java]: Does not `enableRSS` give the idea that it is a method that "enables" RSS?
At least, that is what sounds to me.

**Comment by DaanHoogland** [comment]
@BruceKuiLiu are you still with us? your branch is out off date, can you rebase?

**Comment by BruceKuiLiu** [comment]
@DaanHoogland Yes. I removed all my branches. I need to create a new pull request.

**Comment by DaanHoogland** [comment]
That's fine. at your own time ;)

**Comment by BruceKuiLiu** [comment]
@DaanHoogland #2816

**Comment by DaanHoogland** [comment]
closing in favour of #2816

**Pull request closed without merging**